docs/guides/src/main/server/hostname.adoc

```diff
@@ -5,193 +5,122 @@
 
-Nevertheless there are environments, where Keycloak is deployed behind a proxy/load balancer that terminates TLS completely and the internal requests are done using the unencrypted HTTP protocol.
+Most of the time, it should be enough to set the `hostname-admin` option in order to change only the *host* of the administration console URLs:
 
-To be able to work with Keycloak using HTTP for these environments, there is the hidden configuration option `hostname-strict-https=<true/false>`. This option is set to `true` by default for the production mode, and `false` for the development mode.
+<@kc.start parameters="--hostname-admin=<host>"/>
 
-When you need to access Keycloak using HTTP in production mode, for example when you use `proxy=edge` and you want to access the administration console internally using HTTP, you have to set `hostname-strict-https=false`, otherwise a blank page will show up.
+However, if you want to set not only the host but also a scheme, port, and path, you can set the `hostname-admin-url` option:
 
-Keep in mind the recommended approach is to always use HTTPS, and this still is true for external clients.
+<@kc.start parameters="--hostname-admin-url=<scheme>://<host>:<port>/<path>"/>
 
-== Using the hostname in development mode
-You run Keycloak in development mode by using `start-dev`.
-In this mode, the hostname setting is optional.
-When it is omitted, the incoming request headers are used.
+Note that the `hostname-admin` and `hostname-admin-url` are mutually exclusive.
 
-=== Example: Hostname in development mode
-.Keycloak configuration:
-<@kc.startdev parameters=""/>
+To reduce attack surface, the administration endpoints for Keycloak and the Admin Console should not be publicly accessible.
+Therefore, you can secure them by using a reverse proxy.
+For more information about which paths to expose using a reverse proxy, see the <@links.server id="reverseproxy"/> Guide.
 
-.Invoked command:
-[source, bash]
-----
-curl GET "https://localhost:8080/realms/master/.well-known/openid-configuration" | jq .
-----
+== Example Scenarios
+The following are more example scenarios and the corresponding commands for setting up a hostname.
 
-.Result:
-[source, bash]
-----
-# Frontend endpoints: request://request:request -> http://localhost:8080
-# Backend endpoints: request://request:request -> http://localhost:8080
-----
+Note that the `start` command requires setting up TLS. The corresponding options are not shown for example purposes. For more details,
+see <@links.server id="enabletls"/> guide.
 
-In this example of using a curl GET request, the result shows the current OpenID Discovery configuration.
-All base URLS are taken from the incoming request, so `http://localhost:8080` is used for all endpoints.
+=== Example 1: Exposing the server behind a TLS termination proxy
```

Comment:
```suggestion
=== Exposing the server behind a TLS termination proxy
```

Reply:
There's some mistakes in the numbering below, but I don't think it's needed to prefix these titles with "Example #:"